### ISSUE TITLE
[BUG] `harvester.install.wipe_disks=true` option is overzealous

### ISSUE THREAD

**Issue by tserong** [harvester/harvester#6533]
**Describe the bug**
When using `harvester.install.wipe_disks=true`, the installer will wipe all the disks it can find by running `lsblk` and checking for `type=disk`. This unfortunately includes USB storage devices, which means if you're booting of a USB stick and enable this option, you'll end up wiping the USB stick you're trying to install the system from.

**To Reproduce**
1. Boot the installer off a USB device, and at boot edit the grub command line to add `harvester.install.wipe_disks=true`
2. Go through the installation screens, and commence installation
3. Observe that the installation fails immediately after wiping the disks. Something like this:

![image](https://github.com/user-attachments/assets/54f97d40-e997-4f53-a816-be7ba4a6534b)

**Expected behavior**
The disks physically _in_ the system are wiped. Externally attached storage (whether local via USB or remote via network) is untouched.

**Support bundle**
N/A

**Environment**
 - Harvester ISO version: current master (circa 1.4), v1.3.x, v1.2.2
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): any

**Additional context**
Original issue: https://github.com/harvester/harvester/issues/4527

**Comment by ibrokethecloud** [comment]
may be we should skip external and usb/iso devices?

**Comment by irishgordo** [comment]
Also seeing this present on v1.4.0-rc1 just as a mention:

![Screenshot from 2024-10-02 16-17-54](https://github.com/user-attachments/assets/f8c4bfe9-b611-4ab5-a690-a688f3ca6ce9)

**Comment by irishgordo** [comment]
I think as a possible workaround a flow like this might be beneficial:

- once user is met with that message
- hop onto the node directly, either by rancher@ipv4 : rancher/rancher -- or by ctrl+alt+f2
- `sudo su`
- `ps aux | grep -ie "installer"`
- `kill -9`  the processes, 
- `/usr/bin/bash -e ...` ^ the installer setup-installer.sh process that was just killed
- re-step through the process of the installer ...avoids having to wait for the node to POST again / reboot

**Comment by irishgordo** [comment]
@tserong - do you think maybe I could close out:
- https://github.com/harvester/harvester/issues/5825
And just bring it's mention in this?

Basically, wipe_disks/wipe-disks -> it will attempt to wipe the USB disk during a USB ISO Installation.
So if a user, has booted - one time - from a USB stick, but added the kernel arg or config.yaml arg, to wipe-disks... it will always try to wipe the USB stick during install time...
But the user might have intended for it to wipe-disks/wipe_disks on the physical node itself and potentially hoping the installer would ignore wiping the disk of the USB stick since it booted into the live ISO interactive installer from the USB stick.

**Comment by tserong** [comment]
@irishgordo yeah, I think it's fine to duplicate that issue against this one.

**Comment by bk201** [comment]
we can consider wipe disks with Harvester installation.

**Comment by marnixbouhuis** [comment]
For my homelab setup I played around with this option for a bit and I ended up building my own "patched" ISO for the time now.

I'm currently simply ignoring any drives that are connected over USB. This is not perfect since the user could be using drives that are connected over a USB bus for real storage (maybe some SBC, etc... that does this).

A better solution would probably be to identify the drive the installer is booted from and ignoring this one.

Anyways, here is what I changed (in https://github.com/harvester/harvester-installer/blob/master/package/harvester-os/files/usr/sbin/harv-install):
```
Subject: [PATCH] Ignore USB drives when wiping all disks
---
Index: package/harvester-os/files/usr/sbin/harv-install
<+>UTF-8
===================================================================
diff --git a/package/harvester-os/files/usr/sbin/harv-install b/package/harvester-os/files/usr/sbin/harv-install
--- a/package/harvester-os/files/usr/sbin/harv-install	(revision bbbfb7c0e5d71c893329036b9a36f0f322ecdef0)
+++ b/package/harvester-os/files/usr/sbin/harv-install	(revision 5c0d8602d09189b80c4404f2f1a05ba684029ffd)
@@ -28,7 +28,7 @@
     # }
     if [ "$HARVESTER_WIPE_DISKS" == "true" ]; then
         echo "wiping all disks"
-        for disk in $(lsblk -d -n -J -o NAME,TYPE | jq -r '.blockdevices[] | select(.type == "disk") | .name')
+        for disk in $(lsblk -d -n -J -o NAME,TYPE,TRAN | jq -r '.blockdevices[] | select(.type == "disk" and .tran != "usb") | .name')
         do
             sgdisk -Z /dev/$disk
             partprobe -s /dev/$disk

```

**Comment by votdev** [comment]
Verified this bug has been fixed.

The `harvester.install.wipe_disks` option has been removed with https://github.com/harvester/harvester-installer/pull/885/files#diff-6a84a5d8c064b0be0d59d2d26e27e2549fcf932c02014eb3f954bec7e780d7f7L29.

Test Information
----
* Environment: QEMU/KVM VM with 16GiB RAM and 500GiB disk
* Harvester Version: **v1.5.0-rc1**

### Verify Steps
- Prepare and USB stick with the v1.5.0-rc1 ISO.
- Create a new VM with 16GiB RAM and 500GiB disk.
- Use the USB stick as boot device.
- Start the system and edit the Grub command line to add `harvester.install.wipe_disks=true` at boot. Note, the option has been removed from the code so it should not affect anything. The test will ensure the USB device will not be formatted anyway.

https://github.com/user-attachments/assets/783b267c-4e14-4cb7-bcbb-2e9a766472be

- The `Wipe all disks` message is not shown in the log.
![Image](https://github.com/user-attachments/assets/c62bd04e-3785-4b42-ad1d-1d34486577b5)
- The installation is in progress.
![Image](https://github.com/user-attachments/assets/07cfac28-e8ef-4a3c-906c-239330cc88ef)
- The installation was successful.
![Image](https://github.com/user-attachments/assets/6f27aab6-689a-4d5f-afa2-ee9e47b7c006)

The USB device was not formatted.
